lib/scenic/adapters/postgres.rb

```diff
@@ -155,17 +155,36 @@ def create_materialized_view(name, sql_definition, no_data: false)
       # @param no_data [Boolean] Default: false. Set to true to create
       #   materialized view without running the associated query. You will need
       #   to perform a non-concurrent refresh to populate with data.
+      # @param side_by_side [Boolean] Default: false. Set to true to create the
+      #   new version under a different name and atomically swap them, limiting
+      #   downtime at the cost of doubling disk usage
       #
       # @raise [MaterializedViewsNotSupportedError] if the version of Postgres
       #   in use does not support materialized views.
       #
       # @return [void]
-      def update_materialized_view(name, sql_definition, no_data: false)
+      def update_materialized_view(
+        name, sql_definition, no_data: false, side_by_side: false
+      )
         raise_unless_materialized_views_supported
 
-        IndexReapplication.new(connection: connection).on(name) do
-          drop_materialized_view(name)
-          create_materialized_view(name, sql_definition, no_data: no_data)
+        if side_by_side
+          session_id = Time.now.to_i
+          new_name = "#{name}_new_#{session_id}"
```

Comment:
Postgres identifiers are limited to 63 characters. We're removing 15 characters from that allotment already. This will likely be problematic for some view names. What does Rails do when, for example, a generated index name would be too long? We may need to access that functionality (or copy it) here.

Reply:
IIRC they use a truncated hash of the generated components.

The name only matters for debugging, so I'd be fine with doing that. Something like

```ruby
new_name = "#{name}_new_#{session_id}"
if new_name.size > 63
  # session_id is 10c, _new_ is 5c; leave lots of space free
  new_name = "#{Digest::SHA256.hexdigest(name)[...40]}_new_#{session_id}"
end
```